lib/ansible/modules/cloud/vmware/vmware_host_powermgmt_policy.py

```diff
@@ -0,0 +1,210 @@
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    esxi_hostname: '{{ esxi_hostname }}'
+    policy: high-performance
+    validate_certs: no
+  delegate_to: localhost
+
+- name: Set the Power Management Policy of all host systems from cluster to high-performance
+  vmware_host_powermgmt_policy:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    cluster_name: '{{ cluster_name }}'
+    policy: high-performance
+    validate_certs: no
+  delegate_to: localhost
+'''
+
+RETURN = r'''
+result:
+    description: metadata about host system's power policy
+    returned: always
+    type: dict
+    sample: {
+        "esxi01": {
+            "msg": "power policy for host 'esxi01' already configured",
+            "error": "",
+        },
+    }
+'''
+
+try:
+    from pyVmomi import vim, vmodl
+except ImportError:
+    pass
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.vmware import PyVmomi, vmware_argument_spec, wait_for_task, TaskError
+from ansible.module_utils._text import to_native
+
+
+class VmwareHostPowerManagement(PyVmomi):
+    """
+    Class to manage power management policy of an ESXi host system
+    """
+    def __init__(self, module):
+        super(VmwareHostPowerManagement, self).__init__(module)
+        cluster_name = self.params.get('cluster_name')
+        esxi_host_name = self.params.get('esxi_hostname')
+        self.hosts = self.get_all_host_objs(cluster_name=cluster_name, esxi_host_name=esxi_host_name)
+        if not self.hosts:
+            self.module.fail_json(msg="Failed to find host system with given configuration.")
+
+    def ensure(self):
+        """
+        Function to manage power management policy of an ESXi host system
+        """
+        results = dict(changed=False, result=dict())
+        policy = self.params.get('policy')
+        host_change_list = []
+        for host in self.hosts:
+            changed = False
+            results['result'][host.name] = dict(msg='', error='')
+
+            if policy == 'high-performance':
+                policy_key = 1
+                policy_short_name = "static"
+            elif policy == 'balanced':
+                policy_key = 2
+                policy_short_name = "dynamic"
+            elif policy == 'low-power':
+                policy_key = 3
+                policy_short_name = "low"
+            elif policy == 'custom':
+                policy_key = 4
+                policy_short_name = "custom"
+
+            power_system = host.configManager.powerSystem
+
+            # get current power policy
+            power_system_info = power_system.info
+            current_host_power_policy = power_system_info.currentPolicy
+            # current_host_power_policy.key
+            # current_host_power_policy.name
+            # current_host_power_policy.shortName
+            # current_host_power_policy.description
```

Comment:
If these are not used, better to remove the commented variables

Reply:
ok